From 95d5227aa9638244281383b5c611937dc2b2088e Mon Sep 17 00:00:00 2001
From: LuK1337 <priv.luk@gmail.com>
Date: Wed, 15 Aug 2018 20:49:26 +0200
Subject: [PATCH 03/45] SystemUI: Make tablets great again

* This fixes following issues:
  - invalid X translation of NotificationStackScrollLayout
  - too big spacing between QS icons in landscape

Change-Id: Ie3492fa30648185b744e37a47e1a32c8c874dfd6
---
 packages/SystemUI/res/values-sw600dp/dimens.xml             | 3 +++
 .../systemui/statusbar/phone/NotificationPanelView.java     | 3 ++-
 .../statusbar/stack/NotificationStackScrollLayout.java      | 6 +++++-
 3 files changed, 10 insertions(+), 2 deletions(-)

diff --git a/packages/SystemUI/res/values-sw600dp/dimens.xml b/packages/SystemUI/res/values-sw600dp/dimens.xml
index 2dc6525ffdd..edfef7840d2 100644
--- a/packages/SystemUI/res/values-sw600dp/dimens.xml
+++ b/packages/SystemUI/res/values-sw600dp/dimens.xml
@@ -16,6 +16,9 @@
 */
 -->
 <resources>
+    <!-- Width for the spacer, used between QS tiles. -->
+    <dimen name="qs_quick_tile_space_width">0dp</dimen>
+
     <!-- Standard notification width + gravity -->
     <dimen name="notification_panel_width">@dimen/standard_notification_panel_width</dimen>
 
diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/phone/NotificationPanelView.java b/packages/SystemUI/src/com/android/systemui/statusbar/phone/NotificationPanelView.java
index 44aba292b50..a06b6a8a6ad 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/phone/NotificationPanelView.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/phone/NotificationPanelView.java
@@ -2525,7 +2525,8 @@ public class NotificationPanelView extends PanelView implements
      * @param x the x-coordinate the touch event
      */
     protected void updateVerticalPanelPosition(float x) {
-        if (mNotificationStackScroller.getWidth() * 1.75f > getWidth()) {
+        if ((mKeyguardShowing && !mDozing && !mPulsing) ||
+                mNotificationStackScroller.getWidth() * 1.75f > getWidth()) {
             resetVerticalPanelPosition();
             return;
         }
diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/stack/NotificationStackScrollLayout.java b/packages/SystemUI/src/com/android/systemui/statusbar/stack/NotificationStackScrollLayout.java
index 9c26c69045d..2f9ac4e0337 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/stack/NotificationStackScrollLayout.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/stack/NotificationStackScrollLayout.java
@@ -3994,7 +3994,11 @@ public class NotificationStackScrollLayout extends ViewGroup
     }
 
     private void updateAntiBurnInTranslation() {
-        setTranslationX(mAntiBurnInOffsetX * mDarkAmount);
+        float x = mAntiBurnInOffsetX * mDarkAmount;
+        if (!onKeyguard()) {
+            x += getTranslationX();
+        }
+        setTranslationX(x);
     }
 
     /**
-- 
2.17.1

